### PR TITLE
feat(vm): Restore system-constants-generator

### PR DESCRIPTION
**Pull request by Deniallugo**
# What ❔

Restore system-constants-generator

## Why ❔

This crate is rarely used, but it is important to keep it alive: it's used sometimes to tune our fee-related constants.

Making sure that the code at least compiles is the smallest yet essential part of maintenance.

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -2,4 +2,3 @@
 .github/release-please/** @RomanBrodetski @perekopskiy @Deniallugo @popzxc
 **/CHANGELOG.md @RomanBrodetski @perekopskiy @Deniallugo @popzxc
 CODEOWNERS @RomanBrodetski @perekopskiy @Deniallugo @popzxc
-.github/workflows/** @matter-labs/devops
```
Comment (perekopskiy) [inline CODEOWNERS:5]: It's a mirroring artifact, please revert this change

File: core/bin/system-constants-generator/src/utils.rs [hunk 2]
```diff
@@ -1,36 +1,60 @@
...
+    Address, Execute, L1BatchNumber, L1TxCommonData, L2ChainId, MiniblockNumber, Nonce,
+    ProtocolVersionId, StorageKey, Timestamp, Transaction, BOOTLOADER_ADDRESS, H256,
+    SYSTEM_CONTEXT_ADDRESS, SYSTEM_CONTEXT_GAS_PRICE_POSITION, SYSTEM_CONTEXT_TX_ORIGIN_POSITION,
+    U256,
 };
 use zksync_utils::{bytecode::hash_bytecode, bytes_to_be_words, u256_to_h256};
 
 use crate::intrinsic_costs::VmSpentResourcesResult;
 
+struct SpecialBootloaderTracer {
```
Comment (perekopskiy) [inline core/bin/system-constants-generator/src/utils.rs:27]: Please add a comment about what this tracer does

File: core/bin/system-constants-generator/src/utils.rs [hunk 3]
```diff
@@ -135,46 +159,69 @@ pub(super) fn get_l1_txs(number_of_txs: usize) -> (Vec<Transaction>, Vec<Transac
...
-        TxExecutionMode::VerifyExecute,
-    );
+    let system_env = SystemEnv {
+        zk_porter_available: false,
+        version: ProtocolVersionId::latest(),
+        base_system_smart_contracts,
+        gas_limit: BLOCK_GAS_LIMIT,
+        execution_mode: TxExecutionMode::VerifyExecute,
+        default_validation_computational_gas_limit: BLOCK_GAS_LIMIT,
+        chain_id: L2ChainId(270),
```
Comment (perekopskiy) [inline core/bin/system-constants-generator/src/utils.rs:223]: Maybe use `L2ChainId::default()`?

File: core/bin/system-constants-generator/src/utils.rs [hunk 4]
```diff
@@ -135,46 +159,69 @@ pub(super) fn get_l1_txs(number_of_txs: usize) -> (Vec<Transaction>, Vec<Transac
...
-    let mut vm = init_vm_inner(
-        &mut oracle_tools,
-        BlockContextMode::NewBlock(block_context, Default::default()),
-        &block_properties,
-        BLOCK_GAS_LIMIT,
-        &base_system_contract,
-        TxExecutionMode::VerifyExecute,
-    );
+    let system_env = SystemEnv {
+        zk_porter_available: false,
```
Comment (perekopskiy) [inline core/bin/system-constants-generator/src/utils.rs:217]: We have a constant `ZK_PORTER_IS_AVAILABLE` for this case

File: yarn.lock
Comment (perekopskiy) [inline yarn.lock:1]: Was this file changed intentionally?

**Review by perekopskiy — APPROVED** [review]
`yarn.lock` is still changed, otherwise looks good

File: yarn.lock
Comment (StanislavBreadless) [inline yarn.lock:1]: Are you sure there should be changes to yarn.lock? Nothing has changed in terms of JS dependencies

**Review by StanislavBreadless — APPROVED** [review]

**Review by perekopskiy — APPROVED** [review]

**Review by StanislavBreadless — APPROVED** [review]
